components/calendar/src/chinese_based.rs

```diff
@@ -252,19 +257,32 @@ impl<C: ChineseBased + CalendarArithmetic> ChineseBasedDateInner<C> {
             "Day should be in range of u8! Value {month_i64} failed for RD {date:?}"
         );
         let day = day_i64 as u8;
-        C::new_chinese_based_date(year, month, day)
+        let is_leap_year = Self::new_year_is_leap_year(first_day_of_year, None);
+        let leap_month = if is_leap_year {
+            Self::get_leap_month_from_new_year(first_day_of_year)
```

Comment:
Observation: this function, `get_leap_month_from_new_year`, seems to loop over all the month numbers and manually check whether each one is a leap month. Is that a function you investigated for performance optimization? It looks like it's doing a lot of work and calls `new_moon_on_or_after` more times than it seems like it should need to.

Also, we are now eagerly calling this function instead of lazily, which could manifest as a perf impact if the date format does not need the month name (most formats do though).

Reply:
We talked about this a bit and couldn't see an easy path forward.